store.go

```diff
@@ -1503,17 +1623,31 @@ func (s *store) CreateContainer(id string, names []string, image, layer, metadat
 			}
 		}
 	} else {
-		if err := rlstore.startWriting(); err != nil {
-			return nil, err
+		if crlstore == nil {
+			// No container layer store, use the regular one
+			if err := rlstore.startWriting(); err != nil {
+				return nil, err
+			}
+			defer rlstore.stopWriting()
 		}
-		defer rlstore.stopWriting()
+
 		if !options.HostUIDMapping && len(options.UIDMap) == 0 {
 			uidMap = s.uidMap
 		}
 		if !options.HostGIDMapping && len(options.GIDMap) == 0 {
 			gidMap = s.gidMap
 		}
 	}
+
+	if crlstore == nil {
+		crlstore = rlstore // Started writing above
```

Comment:
I’m _seriously_ worried about using the same variable for “the container store, as opposed to the primary”, and “either one, the one where we are going to create the layer”, within a single function — and so huge one. Please use separate variables.